src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -507,12 +515,15 @@ public bool TryGetPendingTransaction(Keccak hash, out Transaction transaction)
         }
 
         // TODO: Ensure that nonce is always valid in case of sending own transactions from different nodes.
-        public UInt256 ReserveOwnTransactionNonce(Address address)
+        public UInt256 ReserveOwnTransactionNonce(Transaction tx)
         {
+            tx.SenderAddress ??= _ecdsa.RecoverAddress(tx);
```

Comment:
Not sure if it belongs here, or maybe more in `TxSender` or `TxSealer`?

Reply:
I only put it there to not make a lot of changes. Usually this happens in the `UnknownSenderFilter` issue is the `SenderAddress` is needed before the filters run so i had to add it somewhere without making a lot of changes. 

I will move it to `TxSealer`